### ISSUE TITLE
Add `mask()` method to `Observation` for legal action masking

### ISSUE THREAD

**Issue by smly** [smly/RiichiEnv#80]
Currently, users need to manually iterate over `obs.legal_actions()` to construct a legal action mask for the policy network. This process is verbose and potentially inefficient in Python loops.

Current Workaround:

```py
def act(self, obs):
        with torch.no_grad():
            feat = np.frombuffer(obs.encode(), dtype=np.float32).reshape(46, 34).copy()
            q_values = self.model(torch.from_numpy(feat).unsqueeze(0).to(self.device))
            action_id_map = {}
            # Manually creating the mask
            mask = torch.from_numpy(np.zeros(82, dtype=np.float32)).to(self.device)
            for legal_action in obs.legal_actions():
                aid = legal_action.encode()
                mask[aid] = 1.0
                action_id_map[aid] = legal_action
            q_values = q_values.masked_fill(mask == 0, -1e9)
            action_selected = q_values.argmax(dim=1).item()
            return action_id_map[action_selected]
```

Add a mask() method to the `Observation` class (in native/src/observation.rs) that returns the `mask` directly as bytes.

* The method should return a `uint8` array (or `float32` if necessary, but `uint8` is preferred for efficiency).
* The implementation would iterate over the internal `_legal_actions`, compute their `encode()` index, and set the corresponding index in the `mask` array to 1.
* This would allow for a much cleaner usage:

```py
# Proposed usage
mask = torch.from_numpy(np.frombuffer(obs.mask(), dtype=np.uint8)).to(self.device)
q_values = q_values.masked_fill(mask == 0, -1e9)
```
